### PR TITLE
Add new backport action which backports directly to release branches

### PR DESCRIPTION
**Pull request by kminehart**
Now with very thorough tests!

### CODE REVIEW

**Comment by kminehart** [comment]
Okay hold on, looks like I accidentally deleted the test file :(

File: backport/backport.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/google/go-github/v50/github"
+)
+
+var semverRegex = regexp.MustCompile(`^(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>x|0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$`)
```
Comment (baldm0mma) [inline backport/backport.go:12]: Are we using this other places? Should we abstract it into it's own file to be used other places?

Comment (kminehart) [reply backport/backport.go:12]: not in this repo, at least not yet.

File: backport/action.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name: Backport
+description: Automatically creates a cherry pick PR
+inputs:
+  token:
+    description: |
```
Comment (baldm0mma) [inline backport/action.yml:5]: is this pipe necessary since it's a single line?

File: backport/action.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    required: false
+runs:
+  using: composite
+  steps:
+  - uses: actions/setup-go@v5
+  - shell: bash
+    env:
+      GITHUB_TOKEN: ${{inputs.token}}
+      INPUT_LABELS_TO_ADD: ${{inputs.labelsToAdd}}
+    run: |
```
Comment (baldm0mma) [inline backport/action.yml:19]: same here?

File: backport/main.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+type Inputs struct {
+	Title                  string
+	Labels                 []*github.Label
+	RemoveDefaultReviewers bool
+}
+
+func GetInputs() Inputs {
+	var (
+		labelsStr                 = githubactions.GetInput("labelsToAdd")
+		removeDefaultReviewersStr = githubactions.GetInput("removeDefaultReviewers")
```
Comment (baldm0mma) [inline backport/main.go:24]: I don't see `removeDefaultReviewers` or `title` defined as inputs in the `backport.yml`. Is this an issue?

File: .github/workflows/backport.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        required: true
+
+jobs:
+  create-backport-pr:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        repository: grafana/grafana-github-actions-go
+        ref: km/backport-to-release-branches
```
Comment (aangelisc) [inline .github/workflows/backport.yml:20]: This `ref` is unneeded once this is merged right?

File: backport/event.go [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+	"github.com/sethvargo/go-githubactions"
+)
+
+func UnmarshalEventData(ctx *githubactions.GitHubContext, v any) error {
+	if ctx.EventPath == "" {
+		return fmt.Errorf("event path is empty")
+	}
+
+	eventData, err := os.ReadFile(ctx.EventPath)
```
Comment (baldm0mma) [inline backport/event.go:16]: Nit: I think there's a more modern way to handle this check -> https://pkg.go.dev/os#IsNotExist
```suggestion
eventData, err := os.ReadFile(ctx.EventPath)
if err != nil && !errors.Is(err, fs.ErrNotExist) {
    return fmt.Errorf("could not read event file: %w", err)
}
```

File: backport/main.go [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+			PullRequestNumber: payload.GetPullRequest().GetNumber(),
+			SourceSHA:         payload.GetPullRequest().GetMergeCommitSHA(),
+			SourceTitle:       payload.GetPullRequest().GetTitle(),
+			SourceBody:        payload.GetPullRequest().GetBody(),
+			Target:            target,
+			Labels:            append(inputs.Labels, payload.GetPullRequest().Labels...),
+			Owner:             owner,
+			Repository:        repo,
+		}
+		pr, err := Backport(ctx, client.PullRequests, client.Issues, client.Issues, NewShellCommandRunner(log), opts)
```
Comment (baldm0mma) [inline backport/main.go:108]: There are 2 `client.Issues` args, should one be the backportclient?

**Review by baldm0mma — APPROVED** [review]
Made a few comments, but LGTM!

File: backport/event.go (hunk 6 above)
Comment (kminehart) [reply backport/event.go:16]: true

File: backport/main.go (hunk 4 above)
Comment (kminehart) [reply backport/main.go:24]: yea i'll have to remove those

**Pull request merged**